polynomial.c

```diff
@@ -1,128 +1,157 @@
-      double powerX = power(x, i);
-
-      //printf("%f ", powerX);
-      answer = answer + coeffArr[i] * powerX;
-   }
-   return answer;
-}
+  int i;
+  double  answer = 0;
+  
+  printf("Running Sequentially...\n");
 
-double chunk(int coeffArr[], double x, int rank, int numProcs)
-{
-   
-   // SEND SIDE 
-
-   int startIndex = (rank * MAX)/numProcs; //divide up array into halves, thirds, fourths, etc. 
-   int endIndex = ((rank + 1) * MAX)/numProcs;
- 
-   //set endIndex to be the last element for the highest rank process 
-   if(rank == (numProcs - 1))
-        endIndex = MAX;
-
-   double localSum = 0;
-   for(int i = startIndex; i < endIndex; i++)
-   {
-      localSum = localSum + power(coeffArr[i], x);
-   }
-   //DEBUG printf("local sum for rank %d is %f \n", rank,localSum);
-   
-   // RECEIVE SIDE 
-   int source = 0; 
-   MPI_Status status;
-   double final = localSum; //so we don't have to send it if rank = 0
-   if(rank == 0){
-      int source; 
-      double sum;
-      final = localSum;
-      for (source = 1; source < numProcs; source++)
-      {
-         MPI_Recv(&sum, 1, MPI_DOUBLE, source, MPI_ANY_TAG, MPI_COMM_WORLD, &status);
-         final = final + sum;
-      }
-      
-   }
-   else{
-      int destination = 0;
-      MPI_Send(&localSum, 1, MPI_DOUBLE, destination, 1, MPI_COMM_WORLD);
-   }
-   return final; 
+  for( i = 0; i < numPolynomials;  i++)
+  {
+    
+    double powerX = power(variable, i);
+
+    //printf("%f ", powerX);
+    answer = answer + coeffArr[i] * powerX;
+  }
+  return answer;
 }
 
-double roundRobin(int coeffArr[], double x)
-{
+double roundRobin(int coeffArr[], int numPolynomials, int rank, int numProcs, double variable) {
+  double localTotal = 0.0;
+  int i;
+
+  if (rank == 0) printf("Running with Round Robin...\n");
+
+  for (i = rank; i < numPolynomials; i += numProcs) {
+    //DEBUG printf("LocalTotal:%f\tCoef:%i\tDegree:%i\tVariable:%f\n", localTotal, coeffArr[i], i, VARIABLE);
+    localTotal += evaluateTerm(coeffArr[i], i, variable);
+    //DEBUG printf("Thread %i Evaluated Term %i, new local total %f\n", rank, i, localTotal);
+  }
+
+  //DEBUG printf("Thread %i\tCompleted - Total %f\n", rank, localTotal);
 
-   return 0;
+  return localTotal;
 }
 
-void initialize(int coeffArr[])
+double chunk(int coeffArr[], int numPolynomials, int rank, int numProcs, double variable)
 {
-   int maxDegree = MAX - 1;
-   int i;
-   for( i = 0; i < maxDegree; i++)
-   {
-      coeffArr[i] = COEFFICIENT;
-   }
+  int startIndex = (rank * numPolynomials)/numProcs; //divide up array into halves, thirds, fourths, etc. 
+  int endIndex = ((rank + 1) * numPolynomials)/numProcs;
+  int i;
+
+  if (rank == 0) printf("Running with Chunk...\n");
+
+  //set endIndex to be the last element for the highest rank process 
+  if(rank == (numProcs - 1))
+      endIndex = numPolynomials;
+
+  double localSum = 0;
+  for(i = startIndex; i < endIndex; i++)
+  {
+    localSum += evaluateTerm(coeffArr[i], i, variable);
```

Comment:
I needed to replace `power` with `evaluateTerm` here - results were coming out as 12500, not 100.